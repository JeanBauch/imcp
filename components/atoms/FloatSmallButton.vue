<template>
	<div class="p-[10px] bg-primary-orange-500 md:hover:bg-primary-bordo-500 md:hover:cursor-pointer transition-colors rounded-full flex item scale-[85%] md:scale-100">
		<img :src="`icons/small-icons-colors/white/${icon}.svg`" :alt="`Icone botÃ£o ${icon}`">
	</div>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
	data () {
		return {
		};
	},
	props: {
		icon: {
			type: String,
			default: 'compartilhar',
			validator: (value: string) => [
				'compartilhar',
				'calendario',
			].includes(value.toLowerCase()),
		},
	},
});
</script>
