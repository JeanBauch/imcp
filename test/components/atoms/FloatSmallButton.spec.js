import { shallowMount } from '@vue/test-utils';

import FloatSmallButton from '~/components/atoms/FloatSmallButton.vue';

describe('Components:atoms:FloatSmallButton', () => {
	describe('Snapshots', () => {
		it('should render component with icon:compartilhar', () => {
			const wrapper = shallowMount(FloatSmallButton, {
				propsData: {
					icon: 'compartilhar',
				},
			});

			expect(wrapper.html()).toMatchSnapshot();
		});
	});
});
