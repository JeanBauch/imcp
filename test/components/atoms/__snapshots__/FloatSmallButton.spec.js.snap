// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components:atoms:FloatSmallButton Snapshots should render component with icon:compartilhar 1`] = `"<div class=\\"p-[10px] bg-primary-orange-500 md:hover:bg-primary-bordo-500 md:hover:cursor-pointer transition-colors rounded-full flex item scale-[85%] md:scale-100\\"><img src=\\"icons/small-icons-colors/white/compartilhar.svg\\" alt=\\"Icone botÃ£o compartilhar\\"></div>"`;
